trunk/src/core/srs_core_autofree.hpp

```diff
@@ -81,4 +81,103 @@ class impl_SrsAutoFree
     }
 };
 
+// Shared ptr smart pointer, see https://github.com/ossrs/srs/discussions/3667#discussioncomment-8969107
+// Usage:
+//      SrsSharedPtr<MyClass> ptr(new MyClass());
+//      ptr->do_something();
+//
+//      SrsSharedPtr<MyClass> cp = ptr;
+//      cp->do_something();
+template<class T>
+class SrsSharedPtr
```

Comment:
It seems `SrsSharedPtr` shared similar api with `std::shared_ptr`, that's good, so why not add a compiling option to control whether use a template alias from `std::shared_ptr` or self defined `shared ptr`.

```suggestion
#ifdef _USE_STD_SMART_PTR
template <class T>
using SrsSharedPtr = std::shared_ptr<T>;
#else
template <class T>
class SrsSharedPtr { 
// current definition 
};
#endif
```
The benefit is to quickly workaround the shared ptr bugs in the future, e.g. the `ref_count_` is not atomic, it's ok for current single-thread coroutine solution, but what if multi-thread coroutine are implemented in the future, then it's disaster. For the basic level infrastructure codes, it's a smart idea to always has a replacement solution.

Reply:
使用 std::shared_ptr 没有带来任何好处，因为如果SrsSharedPtr有bug，我们就必须修复它。